client/tasks/reminder-bar/reminder-bar.tsx

```diff
@@ -0,0 +1,153 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	ONBOARDING_STORE_NAME,
+	OPTIONS_STORE_NAME,
+	TaskListType,
+} from '@woocommerce/data';
+import { Button } from '@wordpress/components';
+import { Link } from '@woocommerce/components';
+import { getAdminLink } from '@woocommerce/settings';
+import { close as closeIcon } from '@wordpress/icons';
+import interpolateComponents from '@automattic/interpolate-components';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import './reminder-bar.scss';
+
+type ReminderBarProps = {
+	taskListId: string;
+	pageTitle: string;
+	updateBodyMargin: () => void;
+};
+
+type ReminderTextProps = {
+	remainingCount: number | null;
+};
+
+const REMINDER_BAR_HIDDEN_OPTION = 'woocommerce_task_list_reminder_bar_hidden';
+
+const ReminderText: React.FC< ReminderTextProps > = ( { remainingCount } ) => {
+	const translationText =
+		remainingCount === 1
+			? /* translators: 1: remaining tasks count */
+			  __(
+					'🎉 Almost there. Only {{strongText}}%1$d step left{{/strongText}} get your store up and running. {{setupLink}}Finish setup{{/setupLink}}',
+					'woocommerce-admin'
+			  )
+			: /* translators: 1: remaining tasks count */
+			  __(
+					"🚀 You're doing great! {{strongText}}%1$d steps left{{/strongText}} to get your store up and running. {{setupLink}}Continue setup{{/setupLink}}",
+					'woocommerce-admin'
+			  );
+
+	return (
+		<p>
+			{ interpolateComponents( {
+				mixedString: sprintf( translationText, remainingCount ),
+				components: {
+					strongText: <strong />,
+					setupLink: (
+						<Link
+							href={ getAdminLink( 'admin.php?page=wc-admin' ) }
+							type="wp-admin"
+						/>
+					),
+				},
+			} ) }
+		</p>
+	);
+};
+
+export const TasksReminderBar: React.FC< ReminderBarProps > = ( {
+	taskListId = 'setup_experiment_1',
```

Comment:
This is currently manually assigned to this list ID here, which will likely change in a subsequent PR.